plone/app/uuid/utils.py

```diff
@@ -30,27 +74,28 @@ def uuidToURL(uuid):
 def uuidToObject(uuid):
     """Given a UUID, attempt to return a content object. Will return
     None if the UUID can't be found.
-    """
 
-    brain = uuidToCatalogBrain(uuid)
-    if brain is None:
-        return None
-
-    return brain.getObject()
+    Note: the user may not be authorized to view the object.
+    It is up to the caller to check this, if needed.
+    """
+    path = uuidToPhysicalPath(uuid)
+    if not path:
+        return
+    site = getSite()
+    if site is None:
+        return
+    return site.unrestrictedTraverse(path)
```

Comment:
This is again a breaking change... What happened to the code that was getting the parent unrestricted and the final path restricted?

Reply:
Ah, right, the original code called `getObject`, and this does check security.
Okay, I have restored getting the parent unrestricted but the final path restricted, possibly raising Unauthorized.

Reply:
I am glad that you pay attention! :-)